src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +19,36 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testSave() {
+        Currency usd = Currency.getInstance("USD");
+        Product product1 = new Product("code", "des", BigDecimal.ZERO, usd, 0, "url");
+        productDao.save(product1);
+    }
+
+    @Test
+    public void testGetProduct() throws NoProductWithSuchIdException {
+        productDao.getProduct(0L);
+        productDao.getProduct(10L);
+        productDao.getProduct(12L);
+    }
```

Comment:
Do these methods without asserts really test anything?
Please add more tests for corner cases: save the product with the same values as existing product, etc.

Reply:
These tests only check if exceptions are thrown.
It's done.